server/src/main/java/org/elasticsearch/action/search/ArraySearchPhaseResults.java

```diff
@@ -22,7 +22,13 @@
 class ArraySearchPhaseResults<Result extends SearchPhaseResult> extends SearchPhaseResults<Result> {
     final AtomicArray<Result> results;
 
-    private final RefCounted refCounted = LeakTracker.wrap(AbstractRefCounted.of(this::doClose));
+    private final AtomicBoolean closed = new AtomicBoolean(false);
+
+    private final Releasable releasable = LeakTracker.wrap(() -> {
```

Comment:
Changed plans on this one. Ref-counting was not appropriate here. We want to have this tracked by the context to be release as a last-resort but we also clearly know when this can be released earlier. So I just made it closable with the double-close noop so we can close it wherever.